pytorch_pretrained_bert/modeling_gpt2.py

```diff
@@ -328,7 +328,8 @@ def __init__(self, model_embeddings_weights, config):
 
     def set_embeddings_weights(self, model_embeddings_weights, predict_special_tokens=True):
         self.predict_special_tokens = predict_special_tokens
-        self.decoder.weight = model_embeddings_weights  # Tied weights
+        # Export to TorchScript can't handle parameter sharing so we are cloning them.
+        self.decoder.weight = nn.Parameter(model_embeddings_weights.clone())  # Tied weights
```

Comment:
This is not actually tying weights. We need to add an option `torchscript` in the global configuration (`PreTrainedConfig`) with default to `False` and only tie_weights when it's `False`.